src/Microsoft.AspNetCore.Mvc.Core/Internal/ObjectMethodExecutor.cs

```diff
@@ -3,15 +3,23 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Linq.Expressions;
 using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Core;
+using Microsoft.Extensions.Internal;
 
 namespace Microsoft.AspNetCore.Mvc.Internal
 {
     public class ObjectMethodExecutor
     {
+        private ActionExecutorAsync _executorAsync;
         private ActionExecutor _executor;
 
+        private static readonly MethodInfo _convertOfTMethod =
+            typeof(ObjectMethodExecutor).GetRuntimeMethods().Single(methodInfo => methodInfo.Name == "Convert");
```

Comment:
typeof() expects a type where as nameof returns string. So this may not compile if I am not missing anything?